doc/json.md

```diff
@@ -0,0 +1,272 @@
+
+### Filter
+
+```json
+{
+  "config": {
+    "filter": {
+      "pattern": {
+        "profile-urls": [
+          "https://xapinet.org/xapi/example-profile/v1"
+        ],
+        "pattern-ids": [
+          "https://xapinet.org/xapi/example-profile/v1/patterns#pattern-1"
+        ]
+      }
+    },
+    ...
+  }
+  ...
+}
+```
+The filter map contains all of the filtering options passed to LRS Pipe. Much like the other sections these fields correspond to the many filter CLI args found on the [options page](options.md). We will cover a few basic scenarios here.
+
+#### Pattern Filtering
+
+In the snippet in the above section you can see how an xAPI Profile and a specific Pattern ID might be provided, analogous to the use of `--pattern-profile-url` and `--pattern-id` args.
+
+#### Template Filtering
+
+```json
+{
+  "config": {
+    "filter": {
+      "template": {
+        "profile-urls": [
+          "https://xapinet.org/xapi/example-profile/v1"
+        ],
+        "template-ids": [
+          "https://xapinet.org/xapi/example-profile/v1/templates#template-1"
+        ]
+      }
+    },
+    ...
+  }
+  ...
+}
+```
+Above you can see how you would apply the same structure but for a template-based filter.
+
+#### Concept Filtering
+
+```json
+{
+  "config": {
+    "filter": {
+      "concept": {
+        "profile-urls": [
+          "https://xapinet.org/xapi/example-profile/v1"
+        ],
+        "concept-types": [
+          "Verb"
+        ],
+        "activity-type-ids": [],
+        "verb-ids": [
+          "https://xapinet.org/xapi/example-profile/v1/concepts#verb-1"
+        ],
+        "attachment-usage-types": []
+      }
+    },
+    ...
+  },
+  ...
+}
+```
+
+Above you can see how you can choose a Profile, what types of concepts to filter, and specific IDs to filter for.
+
+#### JsonPath Filtering
+
+```json
+{
+  "config": {
+    "filter": {
+      "path": {
+        "ensure-paths": [
+          [["result"],["score"],["scaled"]]
+        ],
+        "match-paths": [
+          [[["verb"],["id"]],"http://example.com/verb"]
+        ]
+      }
+    }
+  }
+}
+```
+The above example shows both an `ensure-path` (`$.result.score.scaled`) and `match-path` (`$.verb.id=http://example.com/verb`) filter. Note that in the JSON representation these are not stored as JSONPath syntax and are rather an internal format LRSPipe uses. If this is unintuitive, you can generate the appropriate syntax from JSONPath using the instructions below for Generating JSON Config.
+
+### State
+
+```json
+{
+  "state": {
+    "status": "init",
+    "cursor": "1970-01-01T00:00:00.000000000Z",
+    "source": {
+      "errors": []
+    },
+    "target": {
+      "errors": []
+    },
+    "errors": [],
+    "filter": {}
+  }
+}
+```
+Manipulating state in JSON config can become exceptionally complex and dangerous as this is the data representation of the running state of a job. For that reason we will only cover a few fields here.
```

Comment:
Is it mentioned that this will be completely ignored if there is state present in persistent storage?

Reply:
done